### PR TITLE
[FIXED JENKINS-32007][FIXED JENKINS-34242] Use a custom control to display spinner and errors

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-32007](https://issues.jenkins-ci.org/browse/JENKINS-32007) and [JENKINS-34242](https://issues.jenkins-ci.org/browse/JENKINS-34242) (core PR will follow under [JENKINS-42443](https://issues.jenkins-ci.org/browse/JENKINS-42443))

jenkins-32007

@reviewbybees

### CODE REVIEW

**Comment by KostyaSha** [comment]
`Owner/Org`

**Comment by KostyaSha** [comment]
> @KostyaSha owner can be a person too, not just an org

Owner is user, org is account. Current naming confusing.

**Comment by stephenc** [comment]
@KostyaSha, perhaps but off topic for this PR

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java [hunk 1]
```diff
@@ -1335,9 +1343,19 @@ public ListBoxModel doFillRepositoryItems(@CheckForNull @AncestorInPath Item con
...
-                        LOGGER.log(Level.WARNING, e.getMessage());
+                        LogRecord lr = new LogRecord(Level.WARNING,
+                                "Exception retrieving the repositories of the organization {0} on {1} with credentials {2}");
+                        lr.setThrown(e);
+                        lr.setParameters(new Object[]{
+                                repoOwner, apiUri,
+                                credentials == null
+                                        ? "anonymous access"
+                                        : CredentialsNameProvider.name(credentials)
+                        });
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java]: Why are you creating a `LogRecord` and then not `log`ging it?

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java [hunk 2]
```diff
@@ -1356,9 +1374,19 @@ public ListBoxModel doFillRepositoryItems(@CheckForNull @AncestorInPath Item con
...
-                        LOGGER.log(Level.WARNING, e.getMessage());
+                        LogRecord lr = new LogRecord(Level.WARNING,
+                                "Exception retrieving the repositories of the user {0} on {1} with credentials {2}");
+                        lr.setThrown(e);
+                        lr.setParameters(new Object[]{
+                                repoOwner, apiUri,
+                                credentials == null
+                                        ? "anonymous access"
+                                        : CredentialsNameProvider.name(credentials)
+                        });
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java]: ditto

File: src/main/java/org/jenkinsci/plugins/github_branch_source/FillErrorResponse.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Override
+    public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node)
+            throws IOException, ServletException {
+        rsp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        rsp.setContentType("text/html;charset=UTF-8");
+        rsp.setHeader("X-Jenkins-Select-Error", clearList ? "clear" : "retain");
+        rsp.getWriter().print(
+                "<div class=" + "error".toLowerCase(Locale.ENGLISH) + "><img src=\'" + req.getContextPath()
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/FillErrorResponse.java]: What is `toLowerCase` accomplishing here?

**Pull request merged**